src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java

```diff
@@ -54,11 +56,19 @@ public ProviderSqlSource(Configuration configuration, Object provider) {
    * @since 3.4.5
    */
   public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod) {
+    this(configuration, provider, mapperType, mapperMethod, new XMLLanguageDriver());
+  }
+
+  /**
+   * @since 3.4.6
+   */
+  public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod, LanguageDriver languageDriver) {
     String providerMethodName;
     try {
       this.configuration = configuration;
-      this.sqlSourceParser = new SqlSourceBuilder(configuration);
+      this.languageDriver = languageDriver;
```

Comment:
@kazuki43zoo In order to be compatible with the other two LanguageDriver, can I `judge languagedriver instanceof XMLLanguageDriver` here? If false, use the default XMLLanguageDriver.